scripts/ubiquiti-discovery.nse

```diff
@@ -0,0 +1,254 @@
+local nmap = require "nmap"
+local shortport = require "shortport"
+local table = require "table"
+local stdnse = require "stdnse"
+local string = require "string"
+local ipOps = require "ipOps"
+
+description = [[
+Extracts information from Ubiquiti networking devices.
+
+This script leverages Ubiquiti's Discovery Service which is enabled by default
+on many products.
+]]
+
+author = {"Tom Sellers"}
+
+license = "Same as Nmap--See https://nmap.org/book/man-legal.html"
+
+categories = {"default", "discovery", "version", "safe"}
+
+---
+-- @usage
+-- nmap -sU -p 10001 --script ubiquiti-discovery.nse <target>
+--
+---
+-- @output
+-- PORT      STATE SERVICE            VERSION
+-- 10001/udp open  ubiquiti-discovery Ubiquiti Discovery Service (ER-X v1.10.7)
+--
+-- | ubiquiti-discovery:
+-- |   uptime_seconds: 84592
+-- |   uptime: 0 days 23:29:52
+-- |   hostname: ubnt-router
+-- |   product: ER-X
+-- |   firmware: EdgeRouter.ER-e50.v1.10.7.5127989.181001.1227
+-- |   version: v1.10.7
+-- |   mac_ip:
+-- |     80:2a:a8:df:a1:63: 192.168.0.1
+-- |     80:2a:a8:df:a1:5e: 55.55.55.55
+-- |   mac_addresses:
+-- |     80:2a:a8:df:a1:63
+-- |_    80:2a:a8:df:a1:5e
+--
+--@xmloutput
+-- <elem key="uptime_seconds">84592</elem>
+-- <elem key="uptime">0 days 23:33:00</elem>
+-- <elem key="hostname">ubnt-router</elem>
+-- <elem key="product">ER-X</elem>
+-- <elem key="firmware">EdgeRouter.ER-e50.v1.10.7.5127989.181001.1227</elem>
+-- <elem key="version">v1.10.7</elem>
+-- <table key="mac_ip">
+--   <elem key="80:2a:a8:df:a1:63">192.168.0.1</elem>
+--   <elem key="80:2a:a8:df:a1:5e">55.55.55.55</elem>
+-- </table>
+-- <table key="mac_addresses">
+--   <elem>80:2a:a8:df:a1:63</elem>
+--   <elem>80:2a:a8:df:a1:5e</elem>
+-- </table>
+
+
+portrule = shortport.port_or_service(10001, "ubiquiti-discovery", "udp", {"open", "open|filtered"})
+
+PROBE_V1 = string.pack("BB I2",
+  0x01, 0x00, -- version, command
+  0x00, 0x00  -- length
+)
+
+---
+-- Converts uptime seconds into a human readable string
+--
+-- E.g. "86518" -> "1 days 00:01:58"
+--
+-- @param uptime number of seconds of uptime
+-- @return formatted uptime string (days, hours, minutes, seconds)
+local function uptime_str(uptime)
+  local uptime_num = tonumber(uptime)
+  if not uptime_num then
+    return nil
+  end
+
+  local d = math.floor(uptime_num / 86400)
```

Comment:
Any reason not to use `//`, instead of the float divide and floor combo?

Reply:
I think I'm missing your point here, can you provide an example?

Reply:
Why not to use `local d = uptime_num // 86400`

Reply:
Ah, because that code is copy-pasta with modifications from `ventrilo-info.nse` and it worked.  Also, I haven't had to use floor division operators in Lua before and had problems finding references to it as an operator despite it being in the 5.3 manual.